### ISSUE TITLE
epub 3 validation error `'Duplicate ID 'idN''`, with sphinx i18n

### ISSUE THREAD

**Issue by hhsprings** [sphinx-doc/sphinx#3002]
rst:

``` rest
As we can see above, the API is pretty straightforward.  The only bit of magic
involves the ``DEFAULT`` section which provides default values for all other
sections [1]_.  Note also that keys in sections are
case-insensitive and stored in lowercase [1]_.
```

If we don't use locale, it's okey:

``` html
<p>As we can see above, the API is pretty straightforward.  The only bit of magic
involves the <code class="docutils literal"><span class="pre">DEFAULT</span></code> section which provides default values for all other
sections <a class="footnote-reference" href="#id14" id="id1">[1]</a>.  Note also that keys in sections are
case-insensitive and stored in lowercase <a class="footnote-reference" href="#id14" id="id2">[1]</a>.</p>
```

But when we use locale, sphinx i18n generates like this:

``` html
<p>上の例からわかるように、API はとても直感的です。唯一の魔術は、 
<code class="docutils literal"><span class="pre">DEFAULT</span></code> セクションが
他の全てのセクションのためのデフォルト値を提供していることです <a class="footnote-reference brackets" href="#id14" id="id2">1</a> 。
 また、セクション内の各キーは大文字小文字を区別せず、
全て小文字で保存されていることにも注意してください <a class="footnote-reference brackets" href="#id14" id="id2">1</a>。</p>
```

[EpubCheck 4.0.1](https://github.com/IDPF/epubcheck/releases/tag/v4.0.1) complains it is error:

```
ERROR(RSC-005): Error while parsing file 'Duplicate ID 'id2''.
```

**Comment by hhsprings** [comment]
Note.
The fact that `[1]_` appears twice is not essential. <s>For example, genindex-all will always have duplication id=`記号` (`msdid: Symbols`) if we use i18n.</s>

**Comment by tk0miya** [comment]
Fixed by #5571.
Thank you for reporting!
